server/leafnode.go

```diff
@@ -467,28 +585,81 @@ func (c *client) processLeafnodeInfo(info *Info) {
 	}
 
 	// Mark that the INFO protocol has been received.
-	// Capture a nonce here.
-	c.flags.set(infoReceived)
-	c.nonce = []byte(info.Nonce)
-	if info.TLSRequired && c.leaf.remote != nil {
-		c.leaf.remote.TLS = true
+	if c.flags.setIfNotSet(infoReceived) {
```

Comment:
For creds rotation we may want to just comment that we may want to have nonces updated at some point.

Reply:
I see.. should I revert this particular change? (set info received and capture nonce). The thing is that as of now, the "async" INFO will not contain a nonce. I send that INFO after the initial connect, so the info is updated just with new LeafNodeURLs but is void of nonce (which is only set on a copy of leafNodeInfo during connect).

Reply:
Maybe just add a note when we dod auto key rotation work.